diff --git a/elastalert/elastalert.py b/elastalert/elastalert.py
index 24b10ced9..3e9a60f6f 100755
--- a/elastalert/elastalert.py
+++ b/elastalert/elastalert.py
@@ -972,8 +972,14 @@ def init_rule(self, new_rule, new=True):
         try:
             self.modify_rule_for_ES5(new_rule)
         except TransportError as e:
-            elastalert_logger.warning('Error connecting to Elasticsearch for rule {}. '
-                                      'The rule has been disabled.'.format(new_rule['name']))
+            if self.disable_rules_on_error:
+                elastalert_logger.warning('Error connecting to Elasticsearch for rule {}. '
+                                          'The rule has been disabled.'.format(new_rule['name']))
+            else:
+                elastalert_logger.warning('Error connecting to Elasticsearch for rule {}. '
+                                          'We will try again next time.'.format(new_rule['name']))
+                # By deleting rule from rule_hashes, load_rule_changes will try to load this rule again in the next round
+                del self.rule_hashes[new_rule['rule_file']]
             self.send_notification_email(exception=e, rule=new_rule)
             return False